### ISSUE TITLE
WaitVid can also take 6 Cylces

### ISSUE THREAD

**Issue by Sh1nyAnd3o3** [davispuh/gear-emu#28]
Hey again I have another issue for you, please correct me if I missunderstood something:
https://github.com/davispuh/gear-emu/blob/c647059be124692e40ee51066085e997e9e43072/Gear/EmulationCore/NativeCog.cs#L606

Here you wrote minimum of 7 cycles, according to the manual (version 1.2) at some frequencies this can also take 6 cycles, is this captured by this ? I do not have the time right now to ensure my thoughts, but I can tell that :
* after video generator fetched the data it takes one (sysclk) cycle to capture that data was fetched:
```
reg cap;
reg [1:0] snc;

always @(posedge vclk or posedge snc[1])
if (snc[1])
	cap <= 1'b0;
else if (new_set)
 // new_set is true if the frame_counter == 1, thus because you are performing an index shift : frame_counter ==0
	cap <= 1'b1;

always @(posedge clk_cog)
if (enable)
 // here the value who tells "I was fetched" named  'cap' gets loaded but because of non blocking it takes 1 (sysclk) cylce to be captured
	snc <= {snc[0], cap};

assign ack			= snc[0]; // the value propagated to the cog
```
* after this there is one write cycle which is only important if you are working with assembly and want to set the WR flag ( to write the addition result of the add operation performed while the waitvid execution)

this are in complete 2 sysclock cycles after the video generator fetched data and not 3...
As I mentioned I did not perform any tests or had enougth time to ensure that my thoughts are right, but I wanted to let you know what I thought.

Thanks :)

**Comment by Sh1nyAnd3o3** [comment]
PS: if you agree with me, I would create a pullrequest after my thesis is finished, thus I have the time to do so, otherwise I would be happy to know why I am wrong ( also because of my thesis ;))

**Comment by gatuno1** [comment]
I think you are right, but I don't know how to make the correction.
I'll wait to you.

**Comment by gatuno1** [comment]
Hi Sh1nyAnd3o3:

I'd like to know if this issue was corrected with your PR#29?

**Comment by Sh1nyAnd3o3** [comment]
No, first I had some changes in the PR, but in the end I was not sure about this, thus I removed them and wanted to take a further look at before merging them.
